### PR TITLE
 Add 'flatten' parameter to API request

### LINKED ISSUE #281: Add parameter to API request to make flatten of properties optional

### PR DESCRIPTION
**Pull request by matthiasschaub**
### Description

Make flatten of GeoJSON properties of Indicators and Reports optional.

### Corresponding issue

Continue work of #247 
Closes #281 

### New or changed dependencies
- None

### Checklist
- [x] I have updated my branch to `main` (e.g. through `git rebase main`)
- [x] My code follows the [style guide](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#style-guide) and was checked with [pre-commit](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CONTRIBUTING.md#pre-commit) before committing
- [x] I have commented my code
- [x] I have added sufficient unit and integration [tests](https://github.com/GIScience/ohsome-quality-analyst/blob/main/docs/development_setup.md#tests)
- [x] I have updated the [CHANGELOG.md](https://github.com/GIScience/ohsome-quality-analyst/blob/main/CHANGELOG.md)

### CODE REVIEW

File: workers/ohsome_quality_analyst/api/api.py [hunk 1]
```diff
@@ -309,23 +325,40 @@ async def list_fid_fields():
...
 
-    Items matching the given pattern (See 'fnmatch.fnmatch') will be deleted from the
-    properties.
+    If properties are flattened pattern matching (See 'fnmatch.fnmatch') is used to
+    delete the item with given key from the properties, else item with given key will
+    be removed.
     """
 
-    def _remove_item_from_properties(properties: dict, pattern: str) -> None:
+    def _remove_item_from_properties_pattern(properties: dict, pattern: str) -> None:
```
Comment (matthiasschaub) [inline workers/ohsome_quality_analyst/api/api.py:338]: Once #309 has been fixed, the way to include/exclude SVG in the response could be reimplemented. For example to not create the SVG in the first place (hand down the parameter to the `create_figure` call).

File: workers/tests/unittests/test_api_request_models.py [hunk 2]
```diff
@@ -31,17 +36,30 @@ def test_base_indicator_invalid(self):
             request_models.BaseIndicator(
                 name="GhsPopComparisonBuildings", include_svg="foo"
             )
+            request_models.BaseIndicator(
+                name="GhsPopComparisonBuildings", include_html="foo"
+            )
+            request_models.BaseIndicator(
+                name="GhsPopComparisonBuildings", flatten="foo"
+            )
```
Comment (joker234) [inline workers/tests/unittests/test_api_request_models.py:44]: This test isn't working in general. The test breaks after the first `ValueError` and doesn't check if *every* single test is raising a `ValueError`. To reproduce this, replace the test cases with a valid and an invalid one — the tests shouldn't be valid:

```python
def test_base_indicator_invalid(self):
    with self.assertRaises(ValueError):
        request_models.BaseIndicator()
        request_models.BaseIndicator(name="GhsPopComparisonBuildings")
```

Maybe create another PR for this (or an issue at least) and fix this later on.

File: workers/tests/unittests/test_api_request_models.py [hunk 3]
```diff
@@ -31,17 +36,30 @@ def test_base_indicator_invalid(self):
...
+        )
 
     def test_base_report_invalid(self):
         with self.assertRaises(ValueError):
             request_models.BaseReport()
             request_models.BaseReport(name="foo")
             request_models.BaseReport(include_svg=True)
             request_models.BaseReport(name="SimpleReport", includeSvg="foo")
+            request_models.BaseReport(name="SimpleReport", includeHtml="foo")
+            request_models.BaseReport(name="SimpleReport", flatten="foo")
```
Comment (joker234) [inline workers/tests/unittests/test_api_request_models.py:62]: see above

File: workers/ohsome_quality_analyst/api/api.py (hunk 1 above)
Comment (joker234) [reply workers/ohsome_quality_analyst/api/api.py:338]: :+1: Please create an issue or PR for that so we don't forget about it after this PR is merged.

Comment (matthiasschaub) [reply workers/ohsome_quality_analyst/api/api.py:338]: https://github.com/GIScience/ohsome-quality-analyst/issues/313

File: workers/tests/unittests/test_api_request_models.py (hunk 2 above)
Comment (matthiasschaub) [reply workers/tests/unittests/test_api_request_models.py:49]: Ouh. Good catch. Thanks. I made a new commit to fix this.

File: workers/tests/unittests/test_api_request_models.py (hunk 3 above)
Comment (matthiasschaub) [reply workers/tests/unittests/test_api_request_models.py:72]: I made a new commit to fix this.

**Review by joker234 — DISMISSED** [review]

**Review by joker234 — DISMISSED** [review]

**Review by joker234 — APPROVED** [review]

**Pull request merged**